ophyd/status.py

```diff
@@ -264,12 +269,47 @@ def __init__(self, positioner, target, *, start_ts=None,
         self.pos = positioner
         self.target = target
         self.start_ts = start_ts
+        self.start_pos = self.pos.position
         self.finish_ts = None
         self.finish_pos = None
 
+        self._watchers = []
+
+        def _notify_watchers(value, *args, **kwargs):
+            # Throw away extra *args, **kwargs from pyepics.
+            self._notify_watchers(value)
+
+        self.pos.subscribe(_notify_watchers)
+
         # call the base class
         super().__init__(positioner, **kwargs)
 
+    def watch(self, func):
+        """
+        Subscribe to notifications about progress. Useful for progress bars.
+
+        Parameters
+        ----------
+        func : callable
+            Expected signature: ``func(fraction_done)``
+        """
+        self._watchers.append(func)
+
+    def _notify_watchers(self, value):
+        if not self._watchers:
```

Comment:
I would just like this take `*args, **kwargs`, subscribe this method and have logic like

```python
if self.done:
    self.pos.clear_sub(self._notify_watchers)
    return 
```